src/myrtlespeech/run/train.py

```diff
@@ -85,5 +85,8 @@ def fit(
             if cb_handler.on_epoch_end():
                 break
 
+        if seq_to_seq.lr_scheduler is not None:
+            seq_to_seq.lr_scheduler.step()
+
```

Comment:
All stage-related changes/updates should happen within the `for stage in stages` for loop to keep them in the same scope.

The existing implementation works however there may be a time where `stages` does not always contain `Stage.TRAIN`; if this later update is made then the `lr_scheduler` may be `step()`ed during a `Stage.EVAL`. 

A possible alternative is:

```python
￼            if is_training and seq_to_seq.lr_scheduler is not None:
￼                seq_to_seq.lr_scheduler.step()

            if cb_handler.on_epoch_end():
￼                break
```

Note that it includes the `is_training` guard and has moved above the `on_epoch_end`. It may be worth creating a callback phase dedicated to the learning rate scheduler but suggest leaving that for now until required.